packages/vscode-extension/src/project/ide.ts

```diff
@@ -6,27 +6,48 @@ import { Utils } from "../utilities/utils";
 import { extensionContext } from "../utilities/extensionContext";
 import { Logger } from "../Logger";
 import { disposeAll } from "../utilities/disposables";
+import { State } from "../common/State";
 import { LaunchConfiguration } from "../common/LaunchConfig";
+import { EventEmitter } from "stream";
+import { RootStateManager } from "./StateManager";
 import { OutputChannelRegistry } from "./OutputChannelRegistry";
 
 interface InitialOptions {
   initialLaunchConfig?: LaunchConfiguration;
 }
 
-export class IDE implements Disposable {
+export class IDE extends EventEmitter implements Disposable {
```

Comment:
I'd strongly prefer if you just had the emitter in a field.
Also, `vscode.EventEmitter` is love.